pkg/operator/ceph/cluster/cluster_external.go

```diff
@@ -145,6 +145,17 @@ func (c *ClusterController) configureExternalCephCluster(cluster *cluster) error
 			return errors.Wrap(err, "failed to create crash collector kubernetes secret")
 		}
 	}
+	// Create exporter secret
+	if !cluster.Spec.Monitoring.MetricsDisabled {
+		if cluster.ClusterInfo.CephCred.Username == client.AdminUsername &&
+			cluster.ClusterInfo.CephCred.Secret != opcontroller.AdminSecretNameKey {
```

Comment:
@parth-gr I think this is so long ago, no need to worry about this since everything would be converted to the new key now.

Reply:
We still run the legacy code, https://github.com/rook/rook/blob/master/pkg/operator/ceph/controller/cluster_info.go#L171
If I get sometime I will like to refactor the code .

Reply:
Thanks parth for clarifying that this could happen due to legacy code still in the python script.